keyring/backends/libsecret.py

```diff
@@ -0,0 +1,103 @@
+from contextlib import closing
+import logging
+
+from ..util import properties
+from ..backend import KeyringBackend
+from ..credentials import SimpleCredential
+from ..errors import (
+    PasswordDeleteError,
+    ExceptionRaisedContext,
+    KeyringLocked,
+)
+
+available = False
+try:
+    import gi
+    gi.require_version('Secret', '1')
+    from gi.repository import Secret
+    available = True
+except ImportError:
+    pass
+
+log = logging.getLogger(__name__)
+
+
+class Keyring(KeyringBackend):
+    """libsecret Keyring"""
+
+    appid = 'Python keyring library'
+    if available:
+        schema = Secret.Schema.new(
+            "org.freedesktop.Secret.Generic",
+            Secret.SchemaFlags.NONE,
+            {
+                "application": Secret.SchemaAttributeType.STRING,
+                "service": Secret.SchemaAttributeType.STRING,
+                "username": Secret.SchemaAttributeType.STRING,
+            }
+        )
+
+    @properties.ClassProperty
+    @classmethod
+    def priority(cls):
+        with ExceptionRaisedContext() as exc:
+            Secret.__name__
+        if exc:
+            raise RuntimeError("libsecret required")
+        return 5
+
+    def get_password(self, service, username):
+        """Get password of the username for the service"""
+        attributes = {
+            "application": self.appid,
+            "service": service,
+            "username": username,
+        }
+        items = Secret.password_search_sync(self.schema, attributes,
+                                            Secret.SearchFlags.UNLOCK, None)
+        for item in items:
+            return item.retrieve_secret_sync().get_text()
```

Comment:
If the collection was locked, and the user refused to unlock it, this will raise

    gi.repository.GLib.Error: secret-error: Cannot get secret of a locked object (2)